aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -50,19 +51,18 @@ void poisson_cuda_kernel(
     at::Tensor& ret,
     const at::Tensor& lambda,
     std::pair<uint64_t, uint64_t> seeds) {
-  at::cuda::CUDA_tensor_apply2<scalar_t, scalar_t>(
-      ret,
-      lambda,
-      [seeds] __device__(
-          scalar_t & ret_val, const scalar_t& lambda) {
-        curandStatePhilox4_32_10_t state;
-        curand_init(
-            seeds.first,
-            blockIdx.x * blockDim.x + threadIdx.x,
-            seeds.second,
-            &state);
-        ret_val = static_cast<scalar_t>(curand_poisson(&state, lambda));
-      });
+  at::TensorIterator iter;
+  iter.add_output(ret);
+  iter.add_input(lambda);
+  iter.build();
+  at::native::distribution_unary_kernel(iter, seeds,
+    [] GPU_LAMBDA (curandStatePhilox4_32_10_t &state, scalar_t lambda) -> scalar_t {
+      #if defined(__CUDA_ARCH__) || defined(__HIP_PLATFORM_HCC__)
+      return static_cast<scalar_t>(curand_poisson(&state, lambda));
+      #else
+      return static_cast<scalar_t>(std::nan(""));  // just to avoid compiler warning
```

Comment:
Wait, when we decided that disabling ROCM support for this operator is good idea. Also, I wondering why it is not triggering any test failures.

Reply:
It does not disable ROCm on this. What it does is: if we are compiling for CUDA device code, or compiling for ROCm, then do the correct thing. Otherwise just return some nonsense to silence compiler warning.